File: javascript/ch5/Right.js
const { Either } = require('./Either');
const { EitherFactory } = require('./EitherFactory');

class Right extends Either {
  map(f) {
    return EitherFactory.fromNullable(f(this._value));
  }

  getOrElse(other) {
    return this._value;
  }

  orElse() {
    return this;
  }

  chain(f) {
    return f(this._value);
  }

  getOrElseThrow() {
    return this._value;
  }

  filter(f) {
    return EitherFactory.fromNullable(f(this._value) ? this._value : null);
  }

  toString() {
    return `Either.Right(${this._value})`;
  }
}

module.exports = {
  Right
};

File: python/python/ch5/Right.py
from .Either import Either
from .EitherFactory import EitherFactory

class Right(Either):
    def map(self, f):
        return EitherFactory.from_nullable(f(self._value))
      
    def get_or_else(self, other=None):
        return self._value
      
    def or_else(self):
        return self
      
    def chain(self, f):
        return f(self._value)
      
    def get_or_else_throw(self):
        return self._value
      
    def filter(self, f):
        return EitherFactory.from_nullable(self._value if f(self._value) else None)
      
    def to_string(self):
        return f'Either.Right({self._value})'